src/test/java/com/LuuQu/medicalclinic/service/DoctorServiceTest.java

```diff
@@ -0,0 +1,210 @@
+        expectedIds.add(1L);
+        expectedIds.add(2L);
+        expectedIds.add(3L);
+
+        DoctorDto result = doctorService.addDoctorFacility(1L, 3L);
+
+        Assertions.assertEquals(doctor.getId(), result.getId());
+        Assertions.assertEquals(doctor.getFacilities().size(), result.getFacilities().size());
+        Set<Long> actualIds = result.getFacilities().stream()
+                                .map(FacilitySimpleDto::getId)
+                                .collect(Collectors.toSet());
+        Assertions.assertTrue(actualIds.containsAll(expectedIds));
+    }
+
+    @Test
+    void removeDoctorFacility_noDoctorInDb_CorrectException() {
+        when(doctorRepository.findById(1L)).thenReturn(Optional.empty());
+
+        var exception = Assertions.assertThrows(IllegalArgumentException.class,
+                () -> doctorService.removeDoctorFacility(1L, 1L));
+
+        Assertions.assertEquals(exception.getMessage(), "Non-existent doctor");
+    }
+
+    @Test
+    void removeDoctorFacility_noFacilityInDb_CorrectException() {
+        when(doctorRepository.findById(1L)).thenReturn(Optional.of(new Doctor()));
+        when(facilityRepository.findById(1L)).thenReturn(Optional.empty());
+
+        var exception = Assertions.assertThrows(IllegalArgumentException.class,
+                () -> doctorService.removeDoctorFacility(1L, 1L));
+
+        Assertions.assertEquals(exception.getMessage(), "Non-existent facility");
+    }
+    @Test
+    void removeDoctorFacility_correctApproach_DtoReturned() {
+        Doctor doctor = doctorMapper.toEntity(getDoctorDto());
+        doctor.setId(1L);
+        addTwoFacilitiesToSet(doctor);
+        Facility facility = getFacility();
+        facility.setId(2L);
+        when(doctorRepository.findById(1L)).thenReturn(Optional.of(doctor));
+        when(facilityRepository.findById(2L)).thenReturn(Optional.of(facility));
+        Set<Long> expectedIds = new HashSet<>();
+        expectedIds.add(1L);
+
+        DoctorDto result = doctorService.removeDoctorFacility(1L, 2L);
+
+        Assertions.assertEquals(doctor.getId(), result.getId());
+        Assertions.assertEquals(doctor.getFacilities().size(), result.getFacilities().size());
+        Set<Long> actualIds = result.getFacilities().stream()
+                .map(FacilitySimpleDto::getId)
+                .collect(Collectors.toSet());
+        Assertions.assertTrue(actualIds.containsAll(expectedIds));
+    }
+    Optional<Doctor> setDoctorOptional() {
+        Doctor doctor = new Doctor();
+        doctor.setId(1L);
+        return Optional.of(doctor);
+    }
+
+    DoctorDto getDoctorDto() {
+        DoctorDto doctorDto = new DoctorDto();
+        doctorDto.setEmail("email");
+        doctorDto.setPassword("password");
+        doctorDto.setFirstName("firstName");
+        doctorDto.setLastName("lastName");
+        doctorDto.setSpecialization("specialization");
+        return doctorDto;
+    }
+    Facility getFacility() {
```

Comment:
moze utworz sobie jakas klase TestDataFactory i statyczne metody, ktore zwracaja to co tutaj